addons/fleet/models/fleet_vehicle.py

```diff
@@ -200,12 +204,18 @@ def _search_get_overdue_contract_reminder(self, operator, value):
 
     @api.model
     def create(self, vals):
+        vehicle_type = self.env['fleet.vehicle.model'].browse(vals.get('model_id')).vehicle_type
+        if vehicle_type == 'bike':
+            vals["doors"] = 0
```

Comment:
Seems hackish. On the other hand, we don't apply the same logic on the write method. 

We could make the doors field a stored/editable/computed field, and modify the value to 5 or 0 according to the vehicle type. ;)